chaco/polygon_plot.py

```diff
@@ -100,9 +100,19 @@ def _render(self, gc, points):
             gc.set_line_dash(self.edge_style_)
             gc.set_fill_color(self.effective_face_color)
 
-            gc.lines(points)
-            gc.close_path()
-            gc.draw_path()
+            # bnds is True where polygons are separated + ensures last is True
+            # indx contains the list of indices that separates the polygons
+            # lines contains a list of lines that represents all polygons
+            bnds = [np.isnan(point[0]) for point in points]
```

Comment:
This is likely to be slow: see https://github.com/enthought/chaco/blob/master/chaco/lineplot.py#L266 for comparison to get an idea of how the line plot finds the runs of non-nan values.

Also this code, and the equivalent code below in `hittest` should probably go into the `gather_points` function so that it can be cached and we can avoid re-computation of the polygons on every draw.